mureader.py

```diff
@@ -0,0 +1,679 @@
+        'ET_MATERIALS':ET_MATERIALS,
+        'ET_MATERIAL':ET_MATERIAL,
+        'ET_TEXTURES':ET_TEXTURES,
+        'ET_MESH_START':ET_MESH_START,
+        'ET_MESH_VERTS':ET_MESH_VERTS,
+        'ET_MESH_UV':ET_MESH_UV,
+        'ET_MESH_UV2':ET_MESH_UV2,
+        'ET_MESH_NORMALS':ET_MESH_NORMALS,
+        'ET_MESH_TANGENTS':ET_MESH_TANGENTS,
+        'ET_MESH_TRIANGLES':ET_MESH_TRIANGLES,
+        'ET_MESH_BONE_WEIGHTS':ET_MESH_BONE_WEIGHTS,
+        'ET_MESH_BIND_POSES':ET_MESH_BIND_POSES,
+        'ET_MESH_END':ET_MESH_END,
+        'ET_LIGHT':ET_LIGHT,
+        'ET_TAG_AND_LAYER':ET_TAG_AND_LAYER,
+        'ET_MESH_COLLIDER2':ET_MESH_COLLIDER2,
+        'ET_SPHERE_COLLIDER2':ET_SPHERE_COLLIDER2,
+        'ET_CAPSULE_COLLIDER2':ET_CAPSULE_COLLIDER2,
+        'ET_BOX_COLLIDER2':ET_BOX_COLLIDER2,
+        'ET_WHEEL_COLLIDER':ET_WHEEL_COLLIDER,
+        'ET_CAMERA':ET_CAMERA,
+    }
+
+    ST_CUSTOM = 0
+    ST_DIFFUSE = 1
+    ST_SPECULAR = 2
+    ST_BUMPED = 3
+    ST_BUMPED_SPECULAR = 4
+    ST_EMISSIVE = 5
+    ST_EMISSIVE_SPECULAR = 6
+    ST_EMISSIVE_BUMPED_SPECULAR = 7
+    ST_ALPHA_CUTOUT = 8
+    ST_ALPHA_CUTOUT_BUMPED = 9
+    ST_ALPHA = 10
+    ST_ALPHA_SPECULAR = 11
+    ST_ALPHA_UNLIT = 12
+    ST_UNLIT = 13
+    SHADER_TYPES = {
+        'ST_CUSTOM':ST_CUSTOM,
+        'ST_DIFFUSE':ST_DIFFUSE,
+        'ST_SPECULAR':ST_SPECULAR,
+        'ST_BUMPED':ST_BUMPED,
+        'ST_BUMPED_SPECULAR':ST_BUMPED_SPECULAR,
+        'ST_EMISSIVE':ST_EMISSIVE,
+        'ST_EMISSIVE_SPECULAR':ST_EMISSIVE_SPECULAR,
+        'ST_EMISSIVE_BUMPED_SPECULAR':ST_EMISSIVE_BUMPED_SPECULAR,
+        'ST_ALPHA_CUTOUT':ST_ALPHA_CUTOUT,
+        'ST_ALPHA_CUTOUT_BUMPED':ST_ALPHA_CUTOUT_BUMPED,
+        'ST_ALPHA':ST_ALPHA,
+        'ST_ALPHA_SPECULAR':ST_ALPHA_SPECULAR,
+        'ST_ALPHA_UNLIT':ST_ALPHA_UNLIT,
+        'ST_UNLIT':ST_UNLIT,
+    }
+    ShaderNames = (
+        "",
+        "KSP/Diffuse",
+        "KSP/Specular",
+        "KSP/Bumped",
+        "KSP/Bumped Specular",
+        "KSP/Emissive/Diffuse",
+        "KSP/Emissive/Specular",
+        "KSP/Emissive/Bumped Specular",
+        "KSP/Alpha/Cutoff",
+        "KSP/Alpha/Cutoff Bumped",
+        "KSP/Alpha/Translucent",
+        "KSP/Alpha/Translucent Specular",
+        "KSP/Alpha/Unlit Transparent",
+        "KSP/Unlit",
+    )
+
+    AT_TRANSFORM = 0
+    AT_MATERIAL = 1
+    AT_LIGHT = 2
+    AT_AUDIO_SOURCE = 3
+    ANIMATION_TYPES = {
+        'AT_TRANSFORM':AT_TRANSFORM,
+        'AT_MATERIAL':AT_MATERIAL,
+        'AT_LIGHT':AT_LIGHT,
+        'AT_AUDIO_SOURCE':AT_AUDIO_SOURCE,
+    }
+
+    TT_TEXTURE = 0
+    TT_NORMAL_MAP = 1
+    TEXTURE_TYPES = {
+        'TT_TEXTURE':TT_TEXTURE,
+        'TT_NORMAL_MAP':TT_NORMAL_MAP,
+    }
+
+class MuTexture:
+    def __init__(self):
+        pass
+    def read(self, mu):
+        #print("MuTexture")
+        self.name = mu.read_string()
+        self.type = mu.read_int()
+        #print("   ", self.name, self.type)
+        return self
```

Comment:
I don't like this style a lot (having an empty init and an instance method returning `self` after some initialization). It looks like `read` is the actual constructor.